src/expr/proof_checker.h

```diff
@@ -36,17 +39,67 @@ class ProofRuleChecker
    * Return the formula that is proven by a proof node with the given id,
    * premises and arguments, or null if such a proof node is not well-formed.
    *
+   * Note that the input/output of this method expects to be terms in *Skolem
+   * form*. To facilitate checking, this method converts to/from witness
+   * form, calling the subprocedure checkInternal below.
+   *
+   * @param id The id of the proof node to check
+   * @param children The premises of the proof node to check. These are nodes
+   * corresponding to the conclusion (ProofNode::getResult) of the children
+   * of the proof node we are checking in Skolem form.
+   * @param args The arguments of the proof node to check
+   * @return The conclusion of the proof node, in Skolem form, if successful or
+   * null if such a proof node is malformed.
+   */
+  Node check(PfRule id,
+             const std::vector<Node>& children,
+             const std::vector<Node>& args);
+  /** Single arg version */
+  Node checkChildrenArg(PfRule id, const std::vector<Node>& children, Node arg);
+  /** No arg version */
+  Node checkChildren(PfRule id, const std::vector<Node>& children);
+  /** Single child only version */
+  Node checkChild(PfRule id, Node child);
+  /** Single argument only version */
+  Node checkArg(PfRule id, Node arg);
+
+  /** mk and node */
+  static Node mkAnd(const std::vector<Node>& a);
+  /** get an index from a node, return false if we fail */
+  static bool getIndex(TNode n, uint32_t& i);
```

Comment:
In retrospect I think it's better to not specialize this for "index". We could be using args for something else. So I think it's better to change to `getConstInt`.

Reply:
I'll call it getUInt32

Reply:
sounds good